libpod/networking_linux.go

```diff
@@ -213,15 +201,14 @@ func (r *Runtime) teardownNetNS(ctr *Container) error {
 		return nil
 	}
 
-	// Because we are using iptables to allow the container to resolve DNS
-	// on per IP address, we also need to try to remove the iptables rule
-	// on cleanup. Remove when https://github.com/containernetworking/plugins/pull/75
-	// is merged.
-	for _, r := range ctr.state.NetworkStatus {
-		for _, ip := range r.IPs {
-			if ip.Address.IP.To4() != nil {
-				iptablesDNS("-D", ip.Address.IP.String())
-			}
+	// Remove firewall rules we added on configuring the container.
+	// Will not be necessary once CNI firewall plugin merges upstream.
+	for _, netStatus := range ctr.state.NetworkStatus {
+		firewallConf := &firewall.FirewallNetConf{
+			PrevResult: netStatus,
+		}
+		if err := r.firewallBackend.Del(firewallConf); err != nil {
+			return errors.Wrapf(err, "error removing firewall rules for container %s", ctr.ID())
```

Comment:
Should we continue to attempt to delete other file wall rules and wrap all of the errors together before returning?

Reply:
Probably more trouble than it's worth - 99% of our deployments will likely only be using one network here

Reply:
ok.